drivers/i2c/i2c_mchp_xec_v2.c

```diff
@@ -263,6 +267,7 @@ static int i2c_xec_reset_config(const struct device *dev)
 	regs->RSHTM = xec_cfg_params[data->speed_id].start_hold_time;
 	regs->DATATM = xec_cfg_params[data->speed_id].data_timing;
 	regs->TMOUTSC = xec_cfg_params[data->speed_id].timeout_scale;
+	regs->IDLSC = xec_cfg_params[data->speed_id].config;
```

Comment:
This seems unrelated to the clock frequency change. Though it seems correct, seems it has been missing for a while. 
Is this a fix for something else?

Reply:
Yes, this change is to program the idle scaling register. It was missing so far.